sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -3422,9 +3433,10 @@ def _UpdateSessionIfRequired(
             if documents.ConsistencyLevel.Session == request_headers[http_constants.HttpHeaders.ConsistencyLevel]:
                 is_session_consistency = True
 
-        if is_session_consistency and self.session:
+        if (is_session_consistency and self.session and http_constants.HttpHeaders.SessionToken in response_headers and
```

Comment:
same QQ: session token should still be captured even if database account is not configured with session consistency